### DISCUSSION TITLE
Error on submitting a drawing response

### DISCUSSION THREAD

**Discussion by keijiota** [jspsych/jsPsych#2840] [Q&A] [answered]
Hello everyone, 

I am planning to get a drawing response using a sketchpad plugin. This plugin works perfectly when there is no background image. 
When I specify a background image on which I draw, I cannot finish the response. And there is an error message stating 
"Uncaught DOMException: Failed to execute 'toDataURL' on 'HTMLCanvasElement': Tainted canvases may not be exported.
    at SketchpadPlugin.end_trial    at HTMLButtonElement."
Do you have any idea of how to avoid this error? Do I need to pre-process an image I use? 

![Screenshot 2022-10-28 191820](https://user-images.githubusercontent.com/36744375/198705327-663a396f-7817-4aef-8e5c-06eea2dc62d5.png)

I would be grateful if someone could help me. 

Many thanks,
Keiji

**Comment by jodeleeuw** [comment, marked as answer]
Hi @keijiota,

This is because you are running the experiment locally on your computer, and there are security flags in place in this environment. Here's a relevant [stack overflow explainer](https://stackoverflow.com/questions/22710627/tainted-canvases-may-not-be-exported).

If you run a local server on your machine this will solve the problem. Try the [Live Server Extension](https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer) if you are programming in VS Code; it is easy to use!

**Comment by keijiota** [comment]
Thank you very much @jodeleeuw 
I see the live server extension indeed solves the issue. 
Is there any way to run the sketchpad on Gorilla? I still see the same issue when running my code on their platform. 
Best,
Keiji

**Reply by jodeleeuw** [reply]
I'm afraid I don't know much about why it wouldn't work on Gorilla. You may have to ask their support group.

**Reply by keijiota** [reply]
Thanks again. I found that a simple way to avoid security flags is to turn off the save_final_image.
